clang/lib/Driver/ToolChains/Flang.cpp

```diff
@@ -342,6 +342,10 @@ void Flang::AddAMDGPUTargetArgs(const ArgList &Args,
     StringRef Val = A->getValue();
     CmdArgs.push_back(Args.MakeArgString("-mcode-object-version=" + Val));
   }
+
+  // Check ROCm path if specified
+  const ToolChain &TC = getToolChain();
+  TC.getDeviceLibs(Args);
```

Comment:
What does this have to do with adding `--rocm-path`? Does `flang` correctly inherit the `RocmInstallationDetector` logic once specified?

Reply:
I need it to invoke  `AMDGPUOpenMPToolChain::getDeviceLibs` to check if specified ROCm path is correct.

Reply:
Doing `clang -v` will show you something like
```
Found HIP installation: /opt/rocm, version 6.0.32831
```
Can we not check that way?

Reply:
Please look at the recent change:
https://github.com/llvm/llvm-project/pull/88190/commits/5b106231ab0f5432ba788035635a0d9d91b22ce0
Is this solution better?